consensus-types/types/payload_requests.go

```diff
@@ -116,16 +116,9 @@ func (n *newPayloadRequest) GetParentBeaconBlockRoot() common.Root {
 	return n.parentBeaconBlockRoot
 }
 
+// GetParentProposerPubkey may return a nil parent pub key. See BuildNewPayloadRequestFromFork
+// to understand how parentProposerPubkey gets populated
 func (n *newPayloadRequest) GetParentProposerPubkey() (*crypto.BLSPubkey, error) {
-	if version.IsBefore(n.GetForkVersion(), version.Electra1()) {
-		return nil, errors.Wrap(
-			ErrForkVersionNotSupported,
-			"parent proposer pubkey not supported in newPayloadRequest before electra",
-		)
-	}
-	if n.parentProposerPubkey == nil {
-		return nil, errors.Wrap(ErrNilValue, "parent proposer pubkey cannot be nil")
-	}
```

Comment:
all of this logic is in `BuildNewPayloadRequestFromFork` and `parentProposerPubkey` is unexported. So the getter can be as simple as possible and let the client have a nil pub key (which is fine we usually just pass it around to EngineAPI)